definitions/scenarios/upgrade_to_katello_nightly.rb

```diff
@@ -0,0 +1,91 @@
+module Scenarios::Katello_Nightly
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          feature(:katello_install) || ForemanMaintain.upgrade_in_progress == 'nightly'
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      'nightly'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Katello nightly'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before upgrading to Katello nightly'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Upgrade steps for Katello nightly'
+      tags :migrations
+      run_strategy :fail_fast
+    end
+
+    def set_context_mapping
+      context.map(:assumeyes, Procedures::Installer::Upgrade => :assumeyes)
+    end
+
+    def compose
+      # The version 3.3 is of relevant Foreman version on which Katello 4.5 released
+      add_step(Procedures::Repositories::Setup.new(:version => 'nightly'))
+      modules_to_enable = ['katello:el8', 'pulpcore:el8']
```

Comment:
In other places you automated the `el8` bit, which will help in the future when we add EL9.